src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs

```diff
@@ -130,5 +135,759 @@ public static bool IsEqual(this StringDictionary @this, StringDictionary other)
+        }
+
+        public static bool IsEqual(this Tuple<int, string, uint, long, double, float> @this, Tuple<int, string, uint, long, double, float> other)
+        {
+            return @this != null &&
+                other != null &&
+                @this.Item1 == other.Item1 &&
+                @this.Item2 == other.Item2 &&
+                @this.Item3 == other.Item3 &&
+                @this.Item4 == other.Item4 &&
+                @this.Item5 == other.Item5 &&
+                @this.Item6 == other.Item6;
+        }
+
+        public static bool IsEqual(this Tuple<int, string, uint, long, double, float, decimal> @this, Tuple<int, string, uint, long, double, float, decimal> other)
+        {
+            return @this != null &&
+                other != null &&
+                @this.Item1 == other.Item1 &&
+                @this.Item2 == other.Item2 &&
+                @this.Item3 == other.Item3 &&
+                @this.Item4 == other.Item4 &&
+                @this.Item5 == other.Item5 &&
+                @this.Item6 == other.Item6 &&
+                @this.Item7 == other.Item7;
+        }
+
+        public static bool IsEqual(this Tuple<int, string, uint, long, double, float, decimal, Tuple<Tuple<int>>> @this, Tuple<int, string, uint, long, double, float, decimal, Tuple<Tuple<int>>> other)
+        {
+            return @this != null &&
+                other != null &&
+                @this.Item1 == other.Item1 &&
+                @this.Item2 == other.Item2 &&
+                @this.Item3 == other.Item3 &&
+                @this.Item4 == other.Item4 &&
+                @this.Item5 == other.Item5 &&
+                @this.Item6 == other.Item6 &&
+                @this.Item7 == other.Item7 &&
+                @this.Rest.Item1.Item1 == other.Rest.Item1.Item1;
+        }
+
+        public static bool IsEqual(this Uri @this, Uri other)
+        {
+            if (@this == null && other == null)
+                return true;
+
+            return @this != null &&
+                other != null &&
+                @this.AbsolutePath == other.AbsolutePath &&
+                @this.AbsoluteUri == other.AbsoluteUri &&
+                @this.LocalPath == other.LocalPath &&
+                @this.Authority == other.Authority &&
+                @this.HostNameType == other.HostNameType &&
+                @this.IsDefaultPort == other.IsDefaultPort &&
+                @this.IsFile == other.IsFile &&
+                @this.IsLoopback == other.IsLoopback &&
+                @this.PathAndQuery == other.PathAndQuery &&
+                @this.Segments.SequenceEqual(other.Segments) &&
+                @this.IsUnc == other.IsUnc &&
+                @this.Host == other.Host &&
+                @this.Port == other.Port &&
+                @this.Query == other.Query &&
+                @this.Fragment == other.Fragment &&
+                @this.Scheme == other.Scheme &&
+                //@this.OriginalString == other.OriginalString && // TODO
```

Comment:
@ViktorHofer have you reverted some of my changes? I have removed this line - we do not serialize this field - @ViktorHofer please check if you haven't reverted more things when reverting